convert2rhel/backup.py

```diff
@@ -341,6 +341,79 @@ def restore(self):
         super(RestorableRpmKey, self).restore()
 
 
+class NewRestorableFile(RestorableChange):
+    def __init__(self, filepath):
+        super(NewRestorableFile, self).__init__()
+        self.filepath = filepath
+
+    def enable(self):
+        """Save current version of a file"""
+        # Prevent multiple backup
+        if self.enabled:
+            return
+
+        loggerinst.info("Backing up %s." % self.filepath)
+        if os.path.isfile(self.filepath):
+            try:
+                loggerinst.debug("Copying %s to %s." % (self.filepath, BACKUP_DIR))
+                shutil.copy2(self.filepath, BACKUP_DIR)
+
+            except (OSError, IOError) as err:
+                # IOError for py2 and OSError for py3
+                loggerinst.critical("Error(%s): %s" % (err.errno, err.strerror))
+        else:
+            loggerinst.info("Can't find %s.", self.filepath)
+
+        # Set the enabled value
+        super(NewRestorableFile, self).enable()
+
+    def restore(self, rollback=True):
+        """Restore a previously backed up file"""
+        # We do not have backup
+        if not self.enabled:
+            loggerinst.info("%s hasn't been backed up." % self.filepath)
+            return
+
+        backup_filepath = os.path.join(BACKUP_DIR, os.path.basename(self.filepath))
+        if rollback:
+            loggerinst.task("Rollback: Restore %s from backup" % self.filepath)
+        else:
+            loggerinst.info("Restoring %s from backup" % self.filepath)
+
+        if not os.path.isfile(backup_filepath):
+            loggerinst.info("%s hasn't been backed up." % self.filepath)
+            return
+
+        try:
+            shutil.copy2(backup_filepath, self.filepath)
+        except (OSError, IOError) as err:
```

Comment:
Lots of paths not being tested here

Reply:
@SpyTec I'm not sure if I understood. Do you mean not tested by unit tests or something else?

Reply:
Yes that's right, L353 L381 L389 L393-394 L399